src/App.jsx

```diff
@@ -0,0 +1,26 @@
+import { Routes, Route } from 'react-router-dom';
+import NavBar from './components/NavBar';
+import Categories from './pages/Categories';
+import Home from './pages/Home';
+import './App.css';
+
+function App() {
+  return (
+    <>
+      <header>
+        <NavBar />
+      </header>
+      <main>
+        <Routes>
+          <Route path="/bookSave/" element={<Home />} />
+          <Route path="/bookSave/categories" element={<Categories />} />
+        </Routes>
```

Comment:
Look at the console of this screen shot below:
<img width="1551" alt="image" src="https://github.com/coker2019/booksave/assets/27709832/0b1487a5-9b6e-47be-a6c4-1c4815b5ca62">

- The routes are having some conflict/errors, the message means you're asking the router to render a route you haven't defined. Kindly fix it.
- The links only changes the navigation URL but the same page is rendered as the image.

<img width="1551" alt="image" src="https://github.com/coker2019/booksave/assets/27709832/0ace8db6-da21-42ba-bb35-b4540cc8dd71">